src/Vanilla/VanillaConnect/VanillaConnect.php

```diff
@@ -0,0 +1,372 @@
+     */
+    const JWT_RESPONSE_HEADER_TEMPLATE = self::JWT_REQUEST_HEADER_TEMPLATE;
+
+    /**
+     * @var String Client identifier.
+     */
+    protected $clientID;
+
+    /**
+     * List of errors that were encountered during the validation process.
+     *
+     * @var array
+     */
+    private $errors = [];
+
+    /**
+     * @var String Secret used to hash the JWT.
+     */
+    protected $secret;
+
+    /**
+     * VanillaConnect constructor.
+     *
+     * @param $clientID
+     * @param $secret
+     */
+    public function __construct($clientID, $secret) {
+        $this->clientID = $clientID;
+        $this->secret = $secret;
+    }
+
+    /**
+     * Extract the client ID from a JWT.
+     *
+     * @throws Exception
+     * @param string $jwt JSON Web Token
+     *
+     * @return string The client ID.
+     */
+    public static function extractClientID($jwt) {
+        $parts = explode('.', $jwt);
+        if (count($parts) !== 3) {
+            throw new Exception('Wrong number of segments.');
+        }
+        if (($header = json_decode(JWT::urlsafeB64Decode($parts[0]), true)) === null) {
+            throw new Exception('Invalid header encoding.');
+        }
+        if (!isset($header['azp']) || $header['azp'] === '') {
+            throw new Exception('Client ID is missing from the JWT header.');
+        }
+
+        return $header['azp'];
+    }
+
+    /**
+     * Extract an item from a JWT's claim.
+     *
+     * @throws Exception
+     * @param string $jwt JSON Web Token
+     * @param string $item The item to extract from the claim.
+     *
+     * @return mixed The value of the item or null.
+     */
+    public static function extractItemFromClaim($jwt, $item) {
+        $parts = explode('.', $jwt);
+        if (count($parts) !== 3) {
+            throw new Exception('Wrong number of segments.');
+        }
+        if (($claim = json_decode(JWT::urlsafeB64Decode($parts[1]), true)) === null) {
+            throw new Exception('Invalid claim encoding.');
+        }
+
+        return isset($claim[$item]) ? $claim[$item] : null;
+    }
+
+    /**
+     * @return string
+     */
+    public function getClientID() {
+        return $this->clientID;
+    }
+
+    /**
+     * Return any errors that occurred.
+     *
+     * @return array
+     */
+    public function getErrors() {
+        return $this->errors;
+    }
+
+    /**
+     * @return string
+     */
+    public function getSecret() {
+        return $this->secret;
+    }
+
+    /**
+     *
+     *
+     * @param string $nonce Nonce token to put in the claim.
+     * @param array $extraClaimItems Any item to add to the claim.
+     * @return string JWT or false on failure.
+     */
+    public function createRequestAuthJWT($nonce, array $extraClaimItems = []) {
+        $authHeader = array_merge(
+            self::JWT_REQUEST_HEADER_TEMPLATE,
+            ['azp' => $this->clientID]
```

Comment:
It is not possible to not set the clientID since it is required by the constructor.
I'll add exception to the constructor to make sure that no one does:

`new VanillaConnect(null, '');`